packages/flutter/lib/src/material/toggle_buttons.dart

```diff
@@ -0,0 +1,1121 @@
+    this._isLastButton,
+    this._textDirection,
+    [RenderBox child]
+  ) : super(child);
+
+  // The width and color of the button's leading side border.
+  BorderSide get leadingBorderSide => _leadingBorderSide;
+  BorderSide _leadingBorderSide;
+  set leadingBorderSide(BorderSide value) {
+    if (_leadingBorderSide == value)
+      return;
+    _leadingBorderSide = value;
+    markNeedsLayout();
+  }
+
+  // The width and color of the button's top and bottom side borders.
+  BorderSide get horizontalBorderSide => _horizontalBorderSide;
+  BorderSide _horizontalBorderSide;
+  set horizontalBorderSide(BorderSide value) {
+    if (_horizontalBorderSide == value)
+      return;
+    _horizontalBorderSide = value;
+    markNeedsLayout();
+  }
+
+  // The width and color of the button's trailing side border.
+  BorderSide get trailingBorderSide => _trailingBorderSide;
+  BorderSide _trailingBorderSide;
+  set trailingBorderSide(BorderSide value) {
+    if (_trailingBorderSide == value)
+      return;
+    _trailingBorderSide = value;
+    markNeedsLayout();
+  }
+
+  // The border radii of each corner of the button.
+  BorderRadius get borderRadius => _borderRadius;
+  BorderRadius _borderRadius;
+  set borderRadius(BorderRadius value) {
+    if (_borderRadius == value)
+      return;
+    _borderRadius = value;
+    markNeedsLayout();
+  }
+
+  // Whether or not this toggle button is the first button in the list.
+  bool get isFirstButton => _isFirstButton;
+  bool _isFirstButton;
+  set isFirstButton(bool value) {
+    if (_isFirstButton == value)
+      return;
+    _isFirstButton = value;
+    markNeedsLayout();
+  }
+
+  // Whether or not this toggle button is the last button in the list.
+  bool get isLastButton => _isLastButton;
+  bool _isLastButton;
+  set isLastButton(bool value) {
+    if (_isLastButton == value)
+      return;
+    _isLastButton = value;
+    markNeedsLayout();
+  }
+
+  // The direction in which text flows for this application.
+  TextDirection get textDirection => _textDirection;
+  TextDirection _textDirection;
+  set textDirection(TextDirection value) {
+    if (_textDirection == value)
+      return;
+    _textDirection = value;
+    markNeedsLayout();
+  }
+
+  static double _maxHeight(RenderBox box, double width) {
+    return box == null ? 0.0 : box.getMaxIntrinsicHeight(width);
+  }
+
+  static double _minWidth(RenderBox box, double height) {
+    return box == null ? 0.0 : box.getMinIntrinsicWidth(height);
+  }
+
+  static double _maxWidth(RenderBox box, double height) {
+    return box == null ? 0.0 : box.getMaxIntrinsicWidth(height);
+  }
+
+  @override
+  double computeDistanceToActualBaseline(TextBaseline baseline) {
+    // The baseline of this widget is the baseline of its child
+    return child.computeDistanceToActualBaseline(baseline) + horizontalBorderSide.width;
+  }
+
+  @override
+  double computeMaxIntrinsicHeight(double width) {
+    return horizontalBorderSide.width +
```

Comment:
This expression and the similar ones below can be one liners.

Reply:
I was originally thinking that having multiple succinct lines was better than having a massive one-liner, since it might be easier to read/parse it if it were multiple lines. 

Primarily driven by [style guide suggestions](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#consider-using--for-short-functions-and-methods)